bgpd/bgp_attr.c

```diff
@@ -2262,6 +2284,35 @@ bgp_attr_ext_communities(struct bgp_attr_parser_args *args)
 	return BGP_ATTR_PARSE_PROCEED;
 }
 
+/* IPv6 Extended Community attribute. */
+static bgp_attr_parse_ret_t
+bgp_attr_ipv6_ext_communities(struct bgp_attr_parser_args *args)
+{
+	struct peer *const peer = args->peer;
+	struct attr *const attr = args->attr;
+	const bgp_size_t length = args->length;
+
+	if (length == 0) {
+		attr->ipv6_ecommunity = NULL;
+		/* Empty extcomm doesn't seem to be invalid per se */
+		return BGP_ATTR_PARSE_PROCEED;
```

Comment:
here should be:
```
return bgp_attr_malformed(args, BGP_NOTIFY_UPDATE_OPT_ATTR_ERR,
					  args->total);
```
because `bgp_attr_malformed` uses treat-as-withdraw and here we use _ignore_. But it should be treat-as-withdraw.